### PR TITLE
Add duration and last_played filters to Smart Playlist

### PR DESCRIPTION
**Pull request by dmoo500**
# What does this implement/fix?

Adds two new filter options to Smart Playlist:
- **Duration filter**: Filter tracks by length using MM:SS time format (e.g., 2:30 to 5:00)
- **Last played filter**: Filter tracks not played within a specified time period (hours, days, weeks, or months)

Both filters work at Python level after SQL query, consistent with existing Smart Playlist architecture.

**Related discussions:**
- https://github.com/orgs/music-assistant/discussions/5649 (Duration/track length filter) ✅ fully implemented
- https://github.com/orgs/music-assistant/discussions/5539 (Not played in X time period) ✅ fully implemented with hours/days/weeks/months support

## Types of changes

- [x] Enhancement to an existing feature — `enhancement`

## Checklist

- [x] The code change is tested and works locally.
- [x] `pre-commit run --all-files` passes.
- [x] `pytest` passes, and tests have been added/updated under `tests/` where applicable.
- [ ] For changes to shared models, the companion PR in `music-assistant/models` is linked.
- [x] For changes affecting the UI, the companion PR in `music-assistant/frontend` is linked.
- [x] I have read and complied with the project's [AI Policy](https://github.com/music-assistant/.github/blob/main/AI_POLICY.md) for any AI-assisted contributions.
- [ ] I have raised a PR against the documentation repository targeting the main or beta branch as appropriate.

### Backend Changes
- Extended `SmartPlaylistRules` dataclass with duration and last_played fields
- Duration: `min_duration` and `max_duration` (in seconds)
- Last played: `last_played_before_value` and `last_played_before_unit` (hours/days/weeks/months)
- Added validation for new fields in `validate_rules()`
- Implemented Python-level filtering with helper functions `_filter_by_duration()` and `_filter_by_last_played()`
- Updated manifest.json with documentation URL and enhanced description

### Filter Behavior
- **Duration**: Filters tracks by length in seconds. Both min and max are optional - leave blank to have open-ended ranges.
- **Last played**: Filters tracks not played within a specified time period. Supports hours, days, weeks, and months. Never-played tracks are always included in the result.

### Testing
- Added 11 unit tests covering serialization, validation, and filtering logic
- Tests verify both fields work independently and together
- Tests cover all time units (hours, days, weeks, months)
- Tests validate edge cases (NULL handling, invalid units, partial field specification)

## Frontend PR
music-assistant/frontend#1998

### CODE REVIEW

File: music_assistant/providers/smart_playlist/__init__.py [hunk 1]
```diff
@@ -750,6 +764,20 @@ async def _apply_seed_post_filters(
...
+            tracks = [t for t in tracks if t.duration and t.duration >= rules.min_duration]
+        if rules.max_duration is not None:
+            tracks = [t for t in tracks if t.duration and t.duration <= rules.max_duration]
+
+        # Apply last_played filter in seed mode
+        if rules.last_played_before_days is not None:
+            threshold_timestamp = int(time.time()) - (rules.last_played_before_days * 86400)
+            tracks = [
+                t for t in tracks if t.last_played == 0 or t.last_played < threshold_timestamp
+            ]
```
[the remark this answers is not in the document]
Comment (dmoo500) [reply music_assistant/providers/smart_playlist/__init__.py]: last_played cannot be None in the database schema. The field is defined as `INTEGER NOT NULL DEFAULT 0` across all track tables (see database.py lines 266, 283, 293, 303, 324, 343, 365, 385, 405). The filter correctly checks `last_played == 0` to identify never-played tracks, and `last_played < threshold_timestamp` to filter by recency. No changes needed.

File: music_assistant/providers/smart_playlist/helpers.py [hunk 2]
```diff
@@ -321,6 +351,22 @@ def validate_rules(rules: SmartPlaylistRules) -> None:
         if at not in valid_album_types:
             msg = f"Invalid excluded_album_types value: {at!r}. Must be one of {sorted(valid_album_types)}"
             raise InvalidDataError(msg)
+    if rules.min_duration is not None and rules.min_duration < 0:
+        msg = f"min_duration must be >= 0, got {rules.min_duration}"
+        raise InvalidDataError(msg)
+    if rules.max_duration is not None and rules.max_duration < 0:
+        msg = f"max_duration must be >= 0, got {rules.max_duration}"
```
[the remark this answers is not in the document]
Comment (dmoo500) [reply music_assistant/providers/smart_playlist/helpers.py:368]: Unit tests added in commit 9b0e8fb57.

File: music_assistant/providers/smart_playlist/__init__.py [hunk 3]
```diff
@@ -671,6 +672,19 @@ async def _evaluate_rules(
                 allowed_album_ids = await self._get_album_ids_for_types(rules.album_types)
                 tracks = self._filter_by_album_ids(tracks, allowed_album_ids)
 
+            # Apply duration filters
+            if rules.min_duration is not None:
+                tracks = [t for t in tracks if t.duration and t.duration >= rules.min_duration]
+            if rules.max_duration is not None:
+                tracks = [t for t in tracks if t.duration and t.duration <= rules.max_duration]
```
[the remark this answers is not in the document]
Comment (dmoo500) [reply music_assistant/providers/smart_playlist/__init__.py]: Unit tests added in commit 9b0e8fb57.

File: music_assistant/providers/smart_playlist/__init__.py [hunk 4]
```diff
@@ -671,6 +672,19 @@ async def _evaluate_rules(
...
+
+            # Apply last_played filter (tracks not played in X days)
+            if rules.last_played_before_days is not None:
+                threshold_timestamp = int(time.time()) - (rules.last_played_before_days * 86400)
+                tracks = [
+                    t for t in tracks if t.last_played == 0 or t.last_played < threshold_timestamp
+                ]
```
[the remark this answers is not in the document]
Comment (dmoo500) [reply music_assistant/providers/smart_playlist/__init__.py]: last_played cannot be None in the database schema. The field is defined as `INTEGER NOT NULL DEFAULT 0` across all track tables (see database.py lines 266, 283, 293, 303, 324, 343, 365, 385, 405). The filter correctly checks `last_played == 0` to identify never-played tracks, and `last_played < threshold_timestamp` to filter by recency. No changes needed.

File: music_assistant/providers/smart_playlist/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmoo500) [reply music_assistant/providers/smart_playlist/__init__.py]: fixed

File: music_assistant/providers/smart_playlist/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dmoo500) [reply music_assistant/providers/smart_playlist/__init__.py]: fixed

**Review by marcelveldt — APPROVED** [review]
Nice QoL improvements again @dmoo500 !
